sdk/lib/cmlib/hiveinit.c

```diff
@@ -403,7 +542,16 @@ HvpGetHiveHeader(IN PHHIVE Hive,
     if (!Result) return NotHive;
 
     /* Do validation */
-    if (!HvpVerifyHiveHeader(BaseBlock)) return NotHive;
+    if (!HvpVerifyHiveHeader(BaseBlock, HFILE_TYPE_PRIMARY))
+    {
+        DPRINT1("HvpVerifyHiveHeader() failed. Try restore.\n");
+        if (!HvRestoreFromLog(Hive, BaseBlock))
```

Comment:
This should be done by the **callers** of `HvpGetHiveHeader()`, in case it has returned the `RecoverHeader` error and the current hive has an attached log (`Hive->Log` is TRUE).